Wrappers/Python/cil/processors/PaganinProcessor.py

```diff
@@ -0,0 +1,405 @@
+        if not isinstance(data, (AcquisitionData)):
+            raise TypeError('Processor only supports AcquisitionData')
+      
+        return True
+        
+    def process(self, out=None):
+
+        data  = self.get_input()
+        self._set_geometry(data.geometry, self.override_geometry)
+
+        if out is None:
+            out = data.geometry.allocate(None)
+
+        if self.pad>0:
+            try:
+                Padder.edge(pad_width={'horizontal': self.pad, 'vertical':self.pad})(data)
+                Padder.edge(pad_width={'horizontal': self.pad, 'vertical':self.pad})(out)
+            except:
+                Padder.edge(pad_width={'horizontal': self.pad})(data)
+                Padder.edge(pad_width={'horizontal': self.pad})(out)
+
+        filter_shape = np.shape(data.get_slice(angle=0).as_array())
+        if data.geometry.dimension == '2D':
+            data = np.expand_dims(data.as_array(),2)
+        else:
+            data = data.as_array()
+
+        filter_shape = np.shape(data.take(indices = 0, axis = out.get_dimension_axis('angle')))
+        self._create_filter(filter_shape[0], filter_shape[1], self.override_filter)
+        scaling_factor = -(1/self.mu)
+        for i in tqdm(range(len(out.geometry.angles))):
+            projection = data.take(indices = i, axis = out.get_dimension_axis('angle'))
+            
+            if self.full_retrieval==True:
+                fI = fft2(self.magnification**2*projection)
+                iffI = ifft2(fI*self.filter)
+                processed_projection = scaling_factor*np.log(iffI)
+            else:
+                fI = fft2(projection)
+                processed_projection = ifft2(fI*self.filter)   
+            out.fill(np.squeeze(processed_projection), angle = i)
+        
+        if self.pad>0:
+            try:
+                Slicer(roi={'horizontal': (self.pad,out.get_dimension_size('horizontal')-self.pad), 'vertical':(self.pad,out.get_dimension_size('vertical')-self.pad)})(out)
+            except:
+                Slicer(roi={'horizontal': (self.pad,out.get_dimension_size('horizontal')-self.pad)})(out)
+
+        return out
+    
+    def set_input(self, dataset):
+        """
+        Set the input data to the processor
+
+        Parameters
+        ----------
+        dataset : AcquisitionData
+            The input AcquisitionData
+        """
+        return super().set_input(dataset)
+        
+    def get_output(self, out=None, override_geometry=None, override_filter=None):
+        r'''
+        Function to get output from the PaganinProcessor
+        
+        Parameters
+        ----------
+        out : DataContainer, optional
+           Fills the referenced DataContainer with the processed data
+
+        override_geometry: dict, optional
+            Geometry parameters to use in the phase retrieval if you want to over-ride values found in `data.geometry`. Specify parameters as a dictionary :code:`{'parameter':value}`\
```

Comment:
These lines are rather long, which reduces readability. Consider breaking on multiple lines to keep within PEP-8 suggestion of 79 char max line width.

Reply:
Good point, the line lengths should be updated throughout now!